### PR TITLE
UP-4866:  Passed arguments in the correct order to PortalPermissionEv…

### PR DESCRIPTION
**Pull request by drewwills**
…aluator.hasPermission() within PermissionsRESTController and PrincipalsRESTController

https://issues.jasig.org/browse/UP-4866

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

It's probably been incorrect from the beginning.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PrincipalsRESTController.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public void setGroupListHelper(IGroupListHelper groupListHelper) {
      * @throws Exception
      */
     @PreAuthorize(
-            "hasPermission('string', 'REST', new org.apereo.portal.spring.security.evaluator.AuthorizableActivity('UP_PERMISSIONS', 'VIEW_PERMISSIONS'))")
+            "hasPermission('ALL', 'java.lang.String', new org.apereo.portal.spring.security.evaluator.AuthorizableActivity('UP_PERMISSIONS', 'VIEW_PERMISSIONS'))")
```
Comment (ChristianMurphy) [inline uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PrincipalsRESTController.java]: Should this be permission `REST`?

Comment (drewwills) [inline uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PrincipalsRESTController.java]: No, it shouldn't.  There's no such permission as `REST`.

Left as `REST`, the API endpoint would continue to be unusable (even for admin).

**Comment by drewwills** [comment]
@ChristianMurphy did you merge this one into rel-4-3-patches as well?  If so, did you resolve the ticket?

**Comment by ChristianMurphy** [comment]
@drewwills This has not been backported to `rel-4-3-patches` yet.

**Pull request merged**